sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.cs

```diff
@@ -23,25 +26,52 @@ public void Execute(GeneratorExecutionContext context)
                 return;
             }
 
-            if (context.SyntaxReceiver is not FunctionMethodSyntaxReceiver receiver || receiver.CandidateMethods.Count == 0)
+            if (context.SyntaxReceiver is not FunctionMethodSyntaxReceiver receiver)
             {
                 return;
             }
 
-            var parser = new Parser(context);
-            var functions = parser.GetFunctions(receiver.CandidateMethods);
+            var entryAssemblyFuncs = GetSymbolsMethodSyntaxes(receiver.CandidateMethods, context);
+            var dependentFuncs = GetDependentAssemblyFunctionsSymbols(context);
+            var allMethods = entryAssemblyFuncs.Concat(dependentFuncs);
 
-            if (functions.Count == 0)
+            if (!allMethods.Any())
             {
                 return;
             }
 
+            var parser = new Parser(context);
+            var functions = parser.GetFunctions(allMethods);
             var shouldIncludeAutoGeneratedAttributes = ShouldIncludeAutoGeneratedAttributes(context);
 
             var text = Emitter.Emit(context, functions, shouldIncludeAutoGeneratedAttributes);
             context.AddSource(Constants.FileNames.GeneratedFunctionExecutor, SourceText.From(text, Encoding.UTF8));
         }
 
+        private IEnumerable<IMethodSymbol> GetSymbolsMethodSyntaxes(List<MethodDeclarationSyntax> methods, GeneratorExecutionContext context)
+        {
+            foreach (MethodDeclarationSyntax method in methods)
+            {
+                var model = context.Compilation.GetSemanticModel(method.SyntaxTree);
+
+                if (FunctionsUtil.IsValidFunctionMethod(context, context.Compilation, model, method))
+                {
+                    IMethodSymbol? methodSymbol = (IMethodSymbol)model.GetDeclaredSymbol(method)!;
+                    yield return methodSymbol;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Collect methods with Function attributes on them from dependent/referenced assemblies.
+        /// </summary>
+        private static IEnumerable<IMethodSymbol> GetDependentAssemblyFunctionsSymbols(GeneratorExecutionContext context)
+        {
+            var visitor = new ReferencedAssemblyMethodVisitor(context.Compilation);
+            visitor.Visit(context.Compilation.SourceModule);
+
+            return visitor.FunctionMethods;
+        }
```

Comment:
nit: new line between methods

```suggestion
        }
 
```

Reply:
Fixed line break.